sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs

```diff
@@ -299,5 +299,44 @@ public async Task SerializesExpectedProperties_DerivedType()
 
             Assert.AreEqual(5, cloudEvent.Data.ToObjectFromJson<DerivedTestPayload>().DerivedProperty);
         }
+
+        [Test]
+        public async Task RespectsPortFromUriSendingCloudEvents()
+        {
+            var mockTransport = new MockTransport((request) =>
+            {
+                Assert.AreEqual(100, request.Uri.Port);
+                return new MockResponse(200);
+            });
+            var options = new EventGridPublisherClientOptions
+            {
+                Transport = mockTransport
+            };
+            EventGridPublisherClient client =
+               new EventGridPublisherClient(
+                   new Uri("https://contoso.com:100/api/events"),
+                   new AzureKeyCredential("fakeKey"),
+                   options);
+            var cloudEvent = new CloudEvent(
+                    "record",
+                    "Microsoft.MockPublisher.TestEvent",
+                    new DerivedTestPayload
```

Comment:
Do we need the extra-fancy payload for this test?

Reply:
probably not, but I didn't see the harm in it.

Reply:
With time it becomes harder to understand what's important for the scenario being tested and what isn't.

Reply:
Well, the test name should help with that, but I see your point.